### DISCUSSION TITLE
Any advice for reading cell data?

### DISCUSSION THREAD

**Discussion by johnkit** [pydata/xarray#8929] [Q&A] [answered]
I have simulation datasets that include cell data. In these datasets, the shape of the coordinate arrays are "not aligned" with the variable arrays. For example, for a temperature array of shape 33x40x133, its coordinate arrays are size 34, 41, and 134. Because the coordinate arrays "conflict" with the data array, the coordinates are dropped when I access the temperature array (per https://docs.xarray.dev/en/latest/user-guide/computation.html#coordinates). I can hard-code some logic to track the coordinates outside the data array, but is there any standard practice for use cases like this?

I can make an example data file if this description isn't clear. Thanks for any help

**Comment by dcherian** [comment, marked as answer]
See https://xgcm.readthedocs.io/en/latest/

**Comment by johnkit** [comment]
thanks. We're exploring XGCM now.
